src/core/ir/eval_http.rs

```diff
@@ -56,7 +71,14 @@ impl<'a, 'ctx, Context: ResolverContextLike + Sync> EvalHttp<'a, 'ctx, Context>
         let ctx = &self.evaluation_ctx;
         let is_get = req.method() == reqwest::Method::GET;
         let dl = &self.data_loader;
-        let response = if is_get && dl.is_some() {
+
+        let response = if *self.http_params.use_batcher && self.http_params.group_by.is_some() {
```

Comment:
`self.http_params.use_batcher` set only when `group_by` is there, so drop that check.